stl/src/vector_algorithms.cpp

```diff
@@ -3626,6 +3626,175 @@ __declspec(noalias) void __stdcall __std_replace_8(
 
 } // extern "C"
 
+namespace {
+    template <class _Ty>
+    void* _Remove_fallback(void* const _First, void* const _Last, void* const _Out, const _Ty _Val) noexcept {
+        _Ty* _Src  = reinterpret_cast<_Ty*>(_First);
+        _Ty* _Dest = reinterpret_cast<_Ty*>(_Out);
+
+        while (_Src != _Last) {
+            if (*_Src != _Val) {
+                *_Dest = *_Src;
+                ++_Dest;
+            }
+
+            ++_Src;
+        }
+
+        return _Dest;
+    }
+
+
+    template <size_t _Size_v, size_t _Size_h>
+    constexpr auto _Make_remove_patterns(const unsigned _Mul, const unsigned _Ew) {
+        struct {
+            uint8_t _Shuf[_Size_v][_Size_h];
+            uint8_t _Size[_Size_v];

```

Comment:
I want to elaborate on the table element bit with decision, though it probably doesn't deserve a code comment.
`_Shuf` is explicitly widened for 32-bit and 64-bit cases using `_mm256_cvtepu8_epi32(_mm_loadu_si64(...));`
`_Size` is implicitly widened for all cases. 

This widening causes some extra instructions for `_Shuf`. For `_Size` apparently it is free (`movzx` instead of `mov`)

I prefer having to widen over having larger tables. Larger tables are consuming more cache, which is hard to notice in a synthetic benchmark, but likely to be noticeable in a realistic program.

Currently the `_Shuf` table, which is the largest of the two, is 128 bytes for 64-bit elements, 1 KiB for 8 and 32 bit elements, 2 KiB for 16 bit elements. Typical L1 cache size is 64 KiB per core.